### ISSUE TITLE
Include permissions

### ISSUE THREAD

**Issue by JN-Jones** [mybb/mybb2#132]
We should start to include permissions. A `canViewForum` has been included but there are several others:
- [x] canPostTopics
- [x] canReply
- [x] canOnlySeeOwnTopics
- [x] canAddPolls
- [x] canEditPolls
- [x] canEditOwnPolls
- [x] canVoteInPolls
- [x] canViewProfiles
- [x] canUseCustomTitle
- [x] canUploadAvatar
- [x] canViewWhosOnline
- [x] canViewMemberlist
- [ ] canViewTopics
- [ ] canOnlyReplyToOwnTopics
- [ ] canUseParsingCode (markdown and mycode)
- [ ] canUseImageCode
- [ ] canUseVideoCode
- [ ] canUseMeCode
- [ ] more existing parsing permissions

Moderation permissions should be included in that PR (eg `canSeeDeletedPosts`).

**Comment by Maxime-BHA** [comment]
@JN-Jones @euantorano As of 5.1.11, Laravel introduces an Authorization system based on abilities using a Gate before requests and a policies implementation, IMHO it'd be great to extend the native Authorization system from Laravel, see [here](https://laravel.com/docs/5.1/authorization)

**Comment by euantorano** [comment]
Yes, we know about this and the plan is to migrate the system to use this. The default implementation in Laravel is slightly lacking, so we'll be expanding upon it significantly.
